### PR TITLE
Support for E73-2G4M08S1E (nrf52833) module

### PR DESCRIPTION
**Pull request by jhbruhn**
As the E73-2G4M08S1C is rather hard to get (actually not at all from my experience), I added support for the E73-2G4M08S1E module, which is based on the nrf52833 MCU: https://www.ebyte.com/en/product-view-news.html?id=1035

It is very similar to the original module, but with half as much RAM and FLASH (but well enough for this device).

### CODE REVIEW

File: code/b-parasite/Makefile [hunk 1]
```diff
@@ -1,17 +1,40 @@
...
+
+ifeq ($(PLATFORM), E73_2G4M08S1E)
+	SRC_FILES += \
+		$(SDK_ROOT)/modules/nrfx/mdk/gcc_startup_nrf52833.S \
+  		$(SDK_ROOT)/modules/nrfx/mdk/system_nrf52833.c
+	TARGETS := nrf52833_xxaa
+	CFLAGS += -DNRF52833_XXAA
+	ASMFLAGS += -DNRF52833_XXAA
+  	LINKER_SCRIPT  := b_parasite_gcc_nrf52833.ld
+HEAP_SIZE := 2048
```
Comment (rbaron) [inline code/b-parasite/Makefile:14]: Would you please indent these two lines and the two ones in the `else` body below?

File: code/b-parasite/Makefile [hunk 2]
```diff
@@ -1,17 +1,40 @@
+# Supported platforms:
+# * E73_2G4M08S1C (nrf2480, default)
+# * E73_2G4M08S1E (nrf2833) 
+PLATFORM ?= E73_2G4M08S1C
```
Comment (rbaron) [inline code/b-parasite/Makefile:4]: For some reason I can't get this default value to kick in. If I don't specify a `PLATFORM`, `make` simply bails out and does not compile anything:

```bash
rbaron@mbp.local:~/dev/parasite/b-parasite/code/b-parasite:<feature/nrf52833✔>
% SDK_ROOT=~/dev/nrf52/sdk/nRF5_SDK_17.0.2_d674dde make clean

rm -rf _build
rbaron@mbp.local:~/dev/parasite/b-parasite/code/b-parasite:<feature/nrf52833✔>
% SDK_ROOT=~/dev/nrf52/sdk/nRF5_SDK_17.0.2_d674dde make

make: Nothing to be done for `default'.
rbaron@mbp.local:~/dev/parasite/b-parasite/code/b-parasite:<feature/nrf52833✔>
% SDK_ROOT=~/dev/nrf52/sdk/nRF5_SDK_17.0.2_d674dde PLATFORM=E73_2G4M08S1C make

mkdir _build
cd _build && mkdir nrf52840_xxaa
Assembling file: gcc_startup_nrf52840.S
Compiling file: system_nrf52840.c
Compiling file: nrf_log_backend_rtt.c
Compiling file: nrf_log_backend_serial.c
Compiling file: nrf_log_backend_uart.c
...
```

Does it work for you?

**Review by rbaron — COMMENTED** [review]
Fantastic work @jhbruhn. Thanks for trying out the E73_2G4M08S1E module. I am glad it worked.

I tested the code using a E73_2G4M08S1C b-parasite and it still works perfectly.

I just left two minor comments. Please let me know what you think.

File: code/b-parasite/Makefile (hunk 2 above)
Comment (rbaron) [reply code/b-parasite/Makefile:4]: Seems like the [CI had the same problem](https://github.com/rbaron/b-parasite/pull/13/checks?check_run_id=3498473009):

```bash
+ export SDK_ROOT=/opt/nRF5_SDK
+ SDK_ROOT=/opt/nRF5_SDK
+ export GNU_INSTALL_ROOT=/usr/bin/
+ GNU_INSTALL_ROOT=/usr/bin/
+ cd /github/workspace/code/b-parasite
+ make clean
rm -rf _build
+ make lint
find src -iname *.h -o -iname *.c | xargs clang-format -n -Werror
+ make
make: Nothing to be done for 'default'.
```

**Comment by rbaron** [comment]
One last question: have you managed to assemble and manually test a b-parasite end-to-end with the E73_2G4M08S1E module?

File: code/b-parasite/Makefile (hunk 2 above)
Comment (jhbruhn) [reply code/b-parasite/Makefile]: You're (both :D) right, it had to do with the whitespace after the equals-sign. It works now after I've removed it and added a comment explaining that.

**Comment by jhbruhn** [comment]
Yes, I've assembled a b-parasite (in fact the version from the Photoresistor PR) and it is successfully transmitting to parasite-scanner.

**Pull request merged**